microsoft-edge/extensions-chromium/publish/sidebar.md

```diff
@@ -0,0 +1,242 @@
+
+Below are the steps to add a sidebar to your extension.
+
+
+<!-- ====================================================================== -->
+## Step 1: Modify the manifest file
+
+Every extension for Microsoft Edge has a JSON-formatted manifest file, named `manifest.json`. The manifest file is the blueprint of your extension.
+
+
+<!-- ------------------------------ -->
+#### Permissions and supported API
+
+A new supported API field, `side_panel`, should be included in the extension manifest file along with the permission `sidePanel`.
+
+`manifest.json`:
+```json
+{
+  ...
+  "side_panel": {
+    "default_path": "sidepanel.html"
+  },
+  "permissions": [
+    "sidePanel"
+  ]
+   ...
+}
+```
+
+<!-- ====================================================================== -->
+## Step 2: Adding functionalities to the sidebar
+
+
+<!-- ------------------------------ -->
+#### Example 1: Default sidebar on every site 
+
+A sidebar can be set as default to show the same extension throughout all the tabs opened in the browser window. Default values will persist across sessions. 
+
+`manifest.json`:
+
+```json
+{
+  "name": "My side panel extension",
+  ...
+  "side_panel": {
+    "default_path": "sidepanel.html"
+  }
+  ...
+}
+```
+
+`sidepanel.html`:
+
+```html
+<!DOCTYPE html>
+<html>
+  <head>
+    <title>My Sidepanel</title>
+  </head>
+  <body>
+    <h1>All sites sidepanel extension</h1>
+    <p>This side panel is enabled on all sites</p>
+  </body>
+</html>
+```
+
+
+<!-- ------------------------------ -->
+#### Example 2: Switching to a different sidebar
+
+Users can switch to a different sidebar when they switch tabs.
+
+The following example sets a welcome sidebar on `runtime.OnInstalled()`, when the user navigates to a different tab, it replaces the welcome page sidebar with the main sidebar.
+
+`service-worker.js`:
+
+```js
+const welcomePage = 'sidepanels/welcome-sp.html';
+const mainPage = 'sidepanels/main-sp.html';
+
+chrome.runtime.onInstalled.addListener(() => {
+  chrome.sidePanel.setOptions({ path: welcomePage });
+});
+
+chrome.tabs.onActivated.addListener(async ({ tabId }) => {
+  const { path } = await chrome.sidePanel.getOptions({ tabId });
+  if (path === welcomePage) {
+    chrome.sidePanel.setOptions({ path: mainPage });
+  }
+});
+```
+
+
+<!-- ------------------------------ -->
+#### Example 3: Enabling sidebar on a specific site
+
+An extension can use`sidepanel.setOptions()` to enable a sidebar on a specific site. This can be a particular website a user is using and would want the same extension to open in the sidebar, once this website is opened.
+
+`service-worker.js`:
+
+```js
+const GOOGLE_ORIGIN = 'https://www.google.com';
+
+chrome.tabs.onUpdated.addListener(async (tabId, info, tab) => {
+  if (!tab.url) return;
+  const url = new URL(tab.url);
+  // Enables the side panel on google.com
```

Comment:
```suggestion
  // Enables the side panel on Bing.com
```

Reply:
Outdated surrounding Markdown.
I incorp'd this change, per latest Files Changed tab
https://github.com/MicrosoftDocs/edge-developer/pull/2648/files